components/brave_wallet/browser/zcash/zcash_wallet_service.cc

```diff
@@ -60,27 +63,53 @@ ZCashWalletService::ZCashWalletService(
   keyring_service_->AddObserver(
       keyring_observer_receiver_.BindNewPipeAndPassRemote());
   complete_manager_ = std::make_unique<ZCashTransactionCompleteManager>(this);
+#if BUILDFLAG(ENABLE_ORCHARD)
+  background_orchard_storage_.emplace(
+      base::ThreadPool::CreateSequencedTaskRunner({base::MayBlock()}),
+      zcash_data_path_.AppendASCII(kOrchardDatabaseName));
+#endif
 }
 
-ZCashWalletService::ZCashWalletService(KeyringService* keyring_service,
+ZCashWalletService::ZCashWalletService(base::FilePath zcash_data_path,
+                                       KeyringService* keyring_service,
                                        std::unique_ptr<ZCashRpc> zcash_rpc)
-    : keyring_service_(keyring_service) {
+    : zcash_data_path_(std::move(zcash_data_path)),
+      keyring_service_(keyring_service) {
+  CHECK_IS_TEST();
```

Comment:
why we need ctor only for tests?

Reply:
To provide mocked services like zcash_rpc